src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceDao.java

```diff
@@ -40,9 +40,13 @@
 
     private static final String sqlBucketRetrieve =
         "SELECT p.id AS project_resource_id, google_project_id, google_project_number, profile_id," +
-            " b.id AS bucket_resource_id, name, region, flightid" +
-            " FROM bucket_resource b JOIN project_resource p ON b.project_resource_id = p.id " +
-            " WHERE b.marked_for_delete = false";
+            " b.id AS bucket_resource_id, name, flightid, sr.region as region " +
+            "FROM bucket_resource b " +
+            "JOIN project_resource p ON b.project_resource_id = p.id, " +
+            "dataset_bucket db, " +
+            "storage_resource sr " +
+            "WHERE b.marked_for_delete = false " +
+            "AND db.bucket_resource_id = b.id AND sr.dataset_id = db.dataset_id AND sr.cloud_resource = 'BUCKET'";
```

Comment:
I wonder if this will break for old buckets. I think the dataset_bucket table was only added back in November - it looks like we back-populated the table, but it potentially was not complete? [PR here](https://github.com/DataBiosphere/jade-data-repo/pull/683/files?file-filters%5B%5D=.properties&file-filters%5B%5D=.xml&file-filters%5B%5D=.yaml#diff-ea6afd408ef79541257879ab02213beb76193649c27137e748b2443a630cac0dR23), and [change-set for back populating here](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/resources/db/changesets/20201024_resourcemanagerrefactor2.yaml#L22). Maybe we could ask if we need to account for "buckets used by the dataset in other projects".

Reply:
Thanks for the context! Saman and I were under the impression that there's only one bucket per dataset. Is this not true? Additionally, don't we enforce a "one project per dataset" requirement? We could query prod to see if a situation like this exists.

Reply:
I've modified the query to use a left join so that we'll still get a bucket row even if the region doesn't exist due to any backfilling issues.